### DISCUSSION TITLE
New Features - Zero-shot or Seed Words?

### DISCUSSION THREAD

**Discussion by francescamcc** [MaartenGr/BERTopic#1662] [Q&A] [answered]
I am curious about the differences between Zero-shot and Seed Word features? Both sound great, and require me to provide info on the 'expected' topics via suggested words/topics. Are there use cases where one of these should be chosen over the other, or would you suggest trying both in models and choosing based on results. Any advice welcome, thanks.

**Comment by linxule** [comment]
> I am curious about the differences between Zero-shot and Seed Word features? Both sound great, and require me to provide info on the 'expected' topics via suggested words/topics. Are there use cases where one of these should be chosen over the other, or would you suggest trying both in models and choosing based on results. Any advice welcome, thanks.

In my project, the use of seed words has resulted in these seed words appearing more often (as keywords) in topic representations. It seems that seed words would affect a wide scope of topics. On the other hand, keywords in "expected topics" seem to only affect topics if they contain the specified set of keywords. Just my two cents and curious as to what insights that @MaartenGr may have on this.

**Reply by MaartenGr** [reply, marked as answer]
There are large differences between the two methods:

[Seed Words](https://maartengr.github.io/BERTopic/getting_started/seed_words/seed_words.html) gives the topic model complete freedom to discover any topic regardless of the seed word you gave it. The only thing seed words does is when a topic is creating, ranking certain words higher. 

[Zero-shot topic modeling](https://maartengr.github.io/BERTopic/getting_started/zeroshot/zeroshot.html), in constrast, is a method that allows you to guide the topic creation towards specific topics that you created before. Compared to seed words, this will actually change the topics that are being created and influence the topic assignment. 

To sum up:

* **seed words** will not affect the clustering process (topic assignment) in any way and only the topic representation
* **zero-shot** will affect both the clustering process (topic assignment) and only indirectly the topic representations (through topic assignment)

Also, I would highly advise reading through the links above as they go in-depth about these functionalities.

**Reply by francescamcc** [reply]
Hi @linxule, thank you for providing your insight! 

Hi @MaartenGr , thank you for clarifying this for me, I understand now. It was the topic assignment vs. topic representation that I hadn't grasped. These are great new updates, thank you!
